src/Type/Php/BcMathStringOrNullReturnTypeExtension.php

```diff
@@ -0,0 +1,121 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\UnaryMinus;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use function in_array;
+
+class BcMathStringOrNullReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array($functionReflection->getName(), ['bcdiv', 'bcmod', 'bcpowmod', 'bcsqrt'], true);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		if ($functionReflection->getName() === 'bcsqrt') {
+			return $this->getTypeForBcSqrt($functionCall, $scope);
+		}
+
+		if ($functionReflection->getName() === 'bcpowmod') {
+			return $this->getTypeForBcPowMod($functionCall, $scope);
+		}
+
+		$secondArgument = $scope->getType($functionCall->args[1]->value);
```

Comment:
This will crash if people call it without the second argument.

Reply:
The `isset` check is at line 39. I will rearrange. Thanks. 

I was not able to reproduce this since the second argument is required though. Got this in the testcase when using `bcmod('10')`.
> The data provider specified for PHPStan\Analyser\NodeScopeResolverTest::testFileAsserts is invalid

Reply:
The provider is invalid exactly because of this code I pointed out. It crashes PHPStan with internal error.